src/Lint.jl

```diff
@@ -350,34 +356,116 @@ function lintdir{T<:AbstractString}(dir::T, ctx::LintContext=LintContext())
+        if code[1] == 'I'
+            etype = "info"
+            etypenumber = 3
+        elseif code[1] == 'W'
+            etype = "warning"
+            etypenumber = 2
+        else
+            etype = "error"
+            etypenumber = 1
+        end
+
+        if style == "standard-linter-v1"
+            if haskey(dict_data,"show_code")
+                if !dict_data["show_code"]
+                    msgtext = "$evar: $txt"
+                end
+            else
+                msgtext = "$code $evar: $txt"
+            end
+            push!(output, Dict("type" => etype,
+                               "text" => msgtext,
+                               "range" => errorrange,
+                               "filePath" => file))
+        elseif style == "vscode"
+            push!(output, Dict("severity" => etypenumber,
+                               "message" => "$evar: $txt",
+                               "range" => errorrange,
+                               "filePath" => file,
+                               "code" => code,
+                               "source" => "Lint.jl"))
+        elseif style == "standard-linter-v2"
+            push!(output, Dict("severity" => etype,
+                               "location" => Dict("file" => file,
+                                                   "position" => errorrange),
+                               "excerpt" => code,
+                               "description" => "$evar: $txt"))
+
+        end
+    end
+    return JSON.json(output)
+end
+
+
+function filtermsgs(msgs,dict_data)
+    if haskey(dict_data,"ignore_warnings")
+        if dict_data["ignore_warnings"]
+            msgs = filter(i -> !iswarning(i), msgs)
+        end
+    end
+    if haskey(dict_data,"ignore_info")
+        if dict_data["ignore_info"]
+            msgs = filter(i -> !isinfo(i), msgs)
+        end
+    end
+    if haskey(dict_data,"ignore_codes")
+        msgs = filter(i -> !(string(i.code) in dict_data["ignore_codes"]), msgs)
+    end
+    return msgs
+end
+
+
+function readandwritethestream(conn,style)
+    if style == "original_behaviour"
+        # println("Connection accepted")
+        # Get file, code length and code
+        file = strip(readline(conn))
+        # println("file: ", file)
+        code_len = parse(Int, strip(readline(conn)))
+        # println("Code bytes: ", code_len)
+        code = Compat.UTF8String(read(conn, code_len))
+        # println("Code received")
+        # Do the linting
+        msgs = lintfile(file, code)
+        # Write response to socket
+        for i in msgs
+            write(conn, string(i))
+            write(conn, "\n")
+        end
+        # Blank line to indicate end of messages
         write(conn, "\n")
+    else
+        dict_data = JSON.parse(conn)
+        msgs = lintfile(dict_data["file"], dict_data["code_str"])
+        msgs = filtermsgs(msgs, dict_data)
+        out = convertmsgtojson(msgs, style, dict_data)
+        write(conn,out)
```

Comment:
Here, we can do `JSON.print(conn, out)`, if `out` hasn't already been converted to JSON (see comment above).